### PR TITLE
nuImages v0.1

### PR DESCRIPTION
**Pull request by holger-nutonomy**
This PR implements version 0.1 of the nuImages dataset. It contains:
- Reorganize readme into nuScenes and nuImages sections.
- requirements.txt split into different products.
- Lazy loading for database table.s
- A method to render images.
- Various list methods to print stats.
- A [Jupyter Notebook tutorial](https://github.com/nutonomy/nuscenes-devkit/blob/nuimages_v0.1/python-sdk/tutorials/nuimages_basics_tutorial.ipynb).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,7 +30,43 @@ Welcome to the devkit of the [nuScenes](https://www.nuscenes.org) dataset.
...
+pip install nuscenes-devkit
+```
+For an advanced installation, see [installation](https://github.com/nutonomy/nuscenes-devkit/blob/master/setup/installation.md) for detailed instructions.
+
+## nuImages
+
+### nuImages setup
+Note that until the release of nuImages v1.0, the files will not be available for public download.
+For the devkit to work you will need to download *all* archives.
+Please unpack the archives to the `/data/sets/images` folder \*without\* overwriting folders that occur in multiple archives.
```
Comment (asha-aptiv) [inline README.md:50]: /data/sets/nuimages

File: python-sdk/nuimages/utils.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        return 0, 0, 0  # Black
+    else:
+        return 255, 158, 0  # Orange
+
+
+def annotation_name(attributes: List[dict],
+                    category_name: str,
+                    with_attributes: bool = False) -> str:
+    """
+    Returns the "name" of an annotation, including or not attribute states.
```
Comment (asha-aptiv) [inline python-sdk/nuimages/utils.py:29]: including or not including attribute states.

File: python-sdk/nuimages/nuimages.py [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+import PIL.Image
+import matplotlib.pyplot as plt
+from matplotlib.axes import Axes
+
+from nuimages.utils import default_color, annotation_name, mask_decode
+
+PYTHON_VERSION = sys.version_info[0]
+
+if not PYTHON_VERSION == 3:
+    raise ValueError("nuScenes dev-kit only supports Python version 3.")
```
Comment (asha-aptiv) [inline python-sdk/nuimages/nuimages.py:23]: nuImages

File: python-sdk/tutorials/nuimages_basics_tutorial.ipynb [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+    "We can change the dataroot parameter if the dataset is installed in a different folder. We can also emit it to use the default setup."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Tables\n",
+    "\n",
+    "As described above, the NuScenes class holds several tables. Each table is a list of records, and each record is a dictionary. For example the first record of the category table is stored at:"
```
Comment (asha-aptiv) [inline python-sdk/tutorials/nuimages_basics_tutorial.ipynb:66]: nuImages

File: python-sdk/tutorials/nuimages_basics_tutorial.ipynb [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+    "Initializing the NuImages instance above was very fast, as we did not actually load the tables. Rather, the class implements lazy loading that overwrites the internal `__getattr__()` function to load a table if it is not already stored in memory. The moment we accessed `category`, we could see the table being loaded from disk. To disable such notifications, just set `verbose=False` when initializing the NuImages object. Furthermore lazy loading can be disabled with `lazy=False`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Indexing\n",
+    "\n",
+    "Since all tables are lists of dictionaries, we can use standard Python oeperations on them. A very common operation is to retrieve a particular record by its token. Since this operation takes linear time, we precompute an index that helps to access a record in constant time.\n",
```
Comment (asha-aptiv) [inline python-sdk/tutorials/nuimages_basics_tutorial.ipynb:109]: operations

File: python-sdk/nuimages/nuimages.py [hunk 6]
```diff
@@ -0,0 +1,342 @@
+# nuScenes dev-kit.
```
[the remark this answers is not in the document]
Comment (holger-nutonomy) [reply python-sdk/nuimages/nuimages.py:1]: We'll still call it nuScenes dev-kit..

File: python-sdk/nuimages/nuimages.py (hunk 3 above)
Comment (holger-nutonomy) [reply python-sdk/nuimages/nuimages.py:23]: Same as above. We'll keep calling it nuScenes devkit

**Comment by holger-nutonomy** [comment]
This branch has now been deleted as it is no longer under development, but the tag can still be accessed: https://github.com/nutonomy/nuscenes-devkit/releases/tag/nuimages_v0.1

**Pull request merged**